Sources/SwiftJavaDocumentation/Documentation.docc/SupportedFeatures.md

```diff
@@ -266,5 +270,41 @@ try (var arena = SwiftArena.ofConfined()) {
 }
 ```
 
+### Protocols
 
+> Note: Protocols are currently only supported in JNI mode.
 
+Protocols are imported as Java `interface`s. For now, we require that all
+concrete types of an interface wrap a Swift instance. In the future, we will add support
+for providing Java-based implementations of interfaces, that you can pass to Java functions.
+
+Consider the following Swift protocol:
+```swift
+protocol Named {
+    var name: String { get }
+
+    func describe() -> String
+}
+```
+will be exported as
+```java
+interface Named extends JNISwiftInstance {
+    public String getName();
+
+    public String describe();
+}
+```
+
+#### Parameters
+Any opaque, existential or generic parameters are imported as Java generics.
+This means that the following function:
+```swift
+func f<S: A & B>(x: S, y: any C, z: some D)
+```
+will be exported as
+```java
+<S extends A & B, T1 extends C, T2 extends D> void f(S x, T1 y, T2 z)   
+```
```

Comment:
```suggestion
```
On the Java side, only `SwiftInstance` implementing types may be passed; so this isn't a way for compatibility with just any arbitrary Java interfaces, but specifically, for allowing passing concrete binding types generated by jextract from Swift types which conform a to a given Swift protocol.
```